src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/Tileset.java

```diff
@@ -198,6 +203,16 @@ public Tileset(
         }
     }
 
+    private void validateLimits(TileMatrixSetLimit limit, GridSubset gridSubset, int zoomLevel) {
+
+        long numTilesHeight = gridSubset.getGridSet().getGrid(zoomLevel).getNumTilesHigh();
+        long numTilesWide = gridSubset.getGridSet().getGrid(zoomLevel).getNumTilesWide();
+        if (limit.getMinTileRow() < 0) limit.setMinTileRow(0);
+        if (limit.getMinTileCol() < 0) limit.setMinTileCol(0);
+        if (limit.getMaxTileRow() > numTilesWide - 1) limit.setMaxTileRow(numTilesWide - 1);
+        if (limit.getMaxTileCol() > numTilesHeight - 1) limit.setMaxTileCol(numTilesHeight - 1);
```

Comment:
I think that columns rely on tilesWide and rows on tilesHeight. Could you please doublecheck?

Reply:
Ohh, Got it. I have switched them!